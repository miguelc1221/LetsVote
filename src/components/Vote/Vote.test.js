import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import VoteContainer, { Vote } from './Vote';

const mockStore = configureMockStore();

const polls = {
  votingPoll: [],
  isLoadingPolls: false,
  isVoting: false,
  pollError: {}
}

test('Vote component should render as expected', () => {
  const store = mockStore({polls});
  const component = shallow(<VoteContainer store={store}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
})
