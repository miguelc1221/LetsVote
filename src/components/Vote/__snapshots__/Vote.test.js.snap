exports[`test Vote component should render as expected 1`] = `
<withRouter(Vote)
  deletePoll={[Function]}
  editPoll={[Function]}
  fetchPolls={[Function]}
  fetchSinglePoll={[Function]}
  filterDeletedPoll={[Function]}
  isLoadingPolls={false}
  isVoting={false}
  pollError={Object {}}
  savePoll={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  votingPoll={Array []} />
`;
